sycl/test/sub_group/shuffle.cpp

```diff
@@ -2,9 +2,10 @@
 // CUDA compilation and runtime do not yet support sub-groups.
 //
 // RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -std=c++14 -D SG_GPU %s -o %t_gpu.out
```

Comment:
I suggest moving check `check` template function to a header file and split `shuffle.cpp` into following parts:
`shuffle.cpp`, `shuffle_fp16.cpp` and probably `shuffle_fp64.cpp`.

Why did you add `-std=c++14`? I don't think `half` data type requires this option.

Reply:
This change is intended to fix subgroup regression after cpu native subgroups enbaled by default and keep consistence with other files, eg. scan.cpp, reduce.cpp. I think  we can refine all these tests later.

Reply:
I don't think it's a big deal to "refine" the test, so I prefer we "refine" them first.

I don't get the answer to this question:
> Why did you add `-std=c++14`? I don't think `half` data type requires this option.

Reply:
just copied from scan.cpp/reduce.cpp

Reply:
> I don't think it's a big deal to "refine" the test, so I prefer we "refine" them first.
> 
> I don't get the answer to this question:
> 
> > Why did you add `-std=c++14`? I don't think `half` data type requires this option.

I'll update them

Reply:
Thanks!